en/guides/codablerouting.md

```diff
@@ -15,172 +15,263 @@ redirect_from:
+Codable Routing was made available in Kitura 2.0, before this release only Raw Routing was available. Kitura 2.0 supports both types of Routing, depending on what your application needs. You can implement both types of routing in your application as well. This offers a lot of flexibility on how you build your server. 
 
-## Building a Create API for POST requests
+For many, Codable Routing feels more 'natural' as you work directly with Swift types.  
+This guide shows you step by step how to set up Codable routes on a Kitura server. 
 
-Building a Codable Route for a POST API to implement storing a ToDo item requires only 4 lines of code. The first step is to register a handler for `POST` requests on “/todos”:
+## Let's build a Server!
 
-```swift
-router.post("/todos", handler: storeHandler)
-```
+### Getting started
+First we need a Kitura server to add our routes to. Kitura has a [command line interface](https://github.com/IBM-Swift/kitura-cli) which provides us with utilities for quickly creating a Kitura server.  
+> ![tip] You can find installation instructions for the command line interface in the [Getting Started](/{{ page.lang }}/starter/gettingstarted.html) section. 
 
-and then provide an implementation of the storeHandler:
+Once we've installed the Kitura command line interface we can open a terminal window and run the following:  
 
-```swift
-func storeHandler(todo: ToDo, completion: (ToDo?, RequestError?) -> Void ) -> Void {
-    todoStore.append(todo)
-    completion(todo, nil)
-}
-```
+1. Create a directory:  
+`mkdir ~/CodableRouting`
+2. Change to the new directory  
+`cd ~/CodableRouting`
+3. Create the Kitura server  
+`kitura init`
+
+`kitura init` downloads, creates and builds a starter Kitura application and will take a couple of minutes to complete. This includes best-practice implementations of health checking and monitoring as well as configuration files to allow easily deployment to a Docker container, a Kubernetes cluster, or the IBM Cloud.
 
-Here the store handler accepts a todo item of type `ToDo` and responds using an asynchronous completion handler. The handler then stores the received todo item in the todoStore and returns with the created todo item in the case of success, and an `Error` in the the case of failure.
+`kitura init` also creates an Xcode project for our server so we can open that now:  
+`open CodableRouting.xcodeproj`
 
-The input parameter and the response parameter in completion handler can be any Swift type you want, as long as it conforms to Codable. For example, we could use the following as our `ToDo` type:
+Now we're ready to start adding our Codable routes!
 
+### Creating the models
+Codable routing works directly with Swift value types. So we need to contruct a value type to use. A good example of this is a struct that conforms to Codable: 
```

Comment:
I don't want to start a sentence with so... how about 
"Codable routing works directly with Swift value types, therefore you need to construct a value type to use."